### PR TITLE
OPC-930 Create cold archive bucket

### PR DESCRIPTION
**Pull request by rute-santos**
Feature documentation:
https://harvard-dce.atlassian.net/wiki/spaces/mh/pages/2977235006/Opencast+Cold+Archive

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1418,6 +1418,7 @@ GEM
     simplecov_json_formatter (0.1.4)
 
 PLATFORMS
+  universal-darwin-22
```
Comment (rute-santos) [inline Gemfile.lock]: Hmm Not sure why this was included. Should I remove it?

File: lib/cluster/s3_archive_bucket.rb [hunk 2]
```diff
@@ -11,18 +11,25 @@ def self.find_or_create(bucket_name)
         lifecycle_configuration: {
           rules: [
             {
-              id: "Rotate to Infrequent Access after 30 days",
+              id: "Move to IA after 30 days, Glacier IR after +60",
```
Comment (rute-santos) [inline lib/cluster/s3_archive_bucket.rb:14]: This reflects the change already made in prod some time ago.

File: lib/cluster/s3_cold_archive_bucket.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
+module Cluster
+  class S3ColdArchiveBucket < S3DistributionBucket
+    def self.find_or_create(bucket_name)
+      bucket = S3Bucket.find_existing(name: bucket_name)
+      return bucket if bucket
+
+      S3Bucket.create(name: bucket_name)
+
+      # Currently, there's no way to specify object minimum size in the filter
```
Comment (rute-santos) [inline lib/cluster/s3_cold_archive_bucket.rb:9]: Reference: https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Types/LifecycleRuleFilter.html

File: Gemfile.lock (hunk 1 above)
Comment (lbjay) [reply Gemfile.lock]: I'm not sure and the whole `Gemfile.lock` doesn't always make sense to me. Platforms is where you list the various OSes the bundle of gems can potentially be installed. When you run `bundler` it'll add your OS to that list. I guess there's some situations where if an OS isn't listed there it'll cause problems? This is per https://www.moncefbelyamani.com/understanding-the-gemfile-lock-file/. I think it's safe to just remove it.

**Review by lbjay — APPROVED** [review]

**Pull request merged**